sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunQcDao.java

```diff
@@ -0,0 +1,173 @@
+  private SequencerPartitionContainerStore sequencerPartitionContainerDao;
+  private CascadeType cascadeType;
+
+  private Session currentSession() {
+    return getSessionFactory().getCurrentSession();
+  }
+
+  @Override
+  public long save(RunQC runQc) throws IOException {
+    long id;
+    if (runQc.getId() == AbstractRun.UNSAVED_ID) {
+      id = (long) currentSession().save(runQc);
+    } else {
+      currentSession().update(runQc);
+      id = runQc.getId();
+    }
+    return id;
+  }
+
+  @Override
+  public RunQC get(long id) throws IOException {
+    return (RunQC) currentSession().get(RunQCImpl.class, id);
+  }
+
+  @Override
+  public RunQC lazyGet(long id) throws IOException {
+    return get(id);
+  }
+
+  @Override
+  public Collection<RunQC> listAll() throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunQCImpl.class);
+    @SuppressWarnings("unchecked")
+    List<RunQC> records = criteria.list();
+    return records;
+  }
+
+  @Override
+  public int count() throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunQCImpl.class);
+    return ((Long) criteria.setProjection(Projections.rowCount()).uniqueResult()).intValue();
+  }
+
+  @Override
+  public boolean remove(RunQC runQc) throws IOException {
+    if (runQc.isDeletable()) {
+      currentSession().delete(runQc);
+      RunQC testIfExists = get(runQc.getId());
+
+      return testIfExists == null;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public Collection<RunQC> listByRunId(long runId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunQCImpl.class);
+    criteria.add(Restrictions.eq("run.id", runId));
+    @SuppressWarnings("unchecked")
+    Collection<RunQC> records = criteria.list();
+    return records;
+  }
+
+  @Override
+  public QcType getRunQcTypeById(long qcTypeId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(QcType.class);
+    criteria.add(Restrictions.eq("qcTarget", "Run"));
+    criteria.add(Restrictions.eq("id", qcTypeId));
+    return (QcType) criteria.uniqueResult();
+  }
+
+  @Override
+  public QcType getRunQcTypeByName(String qcName) throws IOException {
+    Criteria criteria = currentSession().createCriteria(QcType.class);
+    criteria.add(Restrictions.eq("qcTarget", "Run"));
+    criteria.add(Restrictions.eq("name", qcName));
+    return (QcType) criteria.uniqueResult();
+  }
+
+  @Override
+  public Collection<QcType> listAllRunQcTypes() throws IOException {
+    Criteria criteria = currentSession().createCriteria(QcType.class);
+    criteria.add(Restrictions.eq("qcTarget", "Run"));
+    @SuppressWarnings("unchecked")
+    Collection<QcType> records = criteria.list();
+    return records;
+  }
+
+  public List<Partition> listPartitionSelectionsByRunQcId(long runQcId) {
+    Criteria criteria = currentSession().createCriteria(RunQCImpl.class);
+    criteria.add(Restrictions.eqOrIsNull("id", runQcId));
```

Comment:
I don't think it's possible to query for the partitions directly -- RunQC to Partition is a unidirectional relationship.